arwen/contexts/storage.go

```diff
@@ -78,9 +85,17 @@ func (context *storageContext) GetStorage(key []byte) []byte {
 	return value
 }
 
-func (context *storageContext) SetStorage(key []byte, value []byte) int32 {
+func (context *storageContext) isElrondReservedKey(key []byte) bool {
+	return bytes.HasPrefix(key, []byte(context.elrondProtectedKeyPrefix))
+}
+
+func (context *storageContext) SetStorage(key []byte, value []byte) (arwen.StorageStatus, error) {
+	if context.isElrondReservedKey(key) {
+		return arwen.StorageUnchanged, errors.New(arwen.UserErrorStoreElrondReservedKey)
```

Comment:
Create specific error constant in `arwen/errors.go` for this case, and return it here instead of `errors.New(...)`. If you need the message itself in some other place, it's better to take it from within the error, rather than define the message separately then instantiating new errors from it every time.